### PR TITLE
fix(use-values): update values when evaluated values changed

### LINKED ISSUE #809: Options defined through feel expressions aren't dynamic
### Describe the Bug

firefox_H4t38wCPEc

As seen above, changing the isGood variable does not affect the select's values, which is unexpected

### Steps to Reproduce

1. add a checkbox with key `isGood`
2. add a select with options expression `if isGood then [ "good"] else ["bad"]`
3. toggle the checkbox and check the select values
4. nothing changes

### Expected Behavior

The select's values should be dynamically evaluated. 

### Environment

- Host Web modeler saas

### PR DESCRIPTION
**Pull request by pinussilvestrus**
Closes #809

This makes sure we re-calculate the values everytime the evaluated values from a `valuesExpression` changed.

### CODE REVIEW

**Review by Skaiir — CHANGES_REQUESTED** [review]
Unfortunately there are some issues here, cannot change option source to expression without crashing the form editor.

File: packages/form-js-viewer/src/render/hooks/useValuesAsync.js [hunk 1]
```diff
@@ -35,11 +33,8 @@ export default function(field) {
   const [ valuesGetter, setValuesGetter ] = useState({ values: [], error: undefined, state: LOAD_STATES.LOADING });
   const initialData = useService('form')._getState().initialData;
 
-  const evaluatedValues = valuesExpression ? (useExpressionEvaluation(valuesExpression) || []) : [];
-
-  // re-calculate values if evaluated expression changes
-  const previousEvaluatedValues = usePrevious(evaluatedValues, [], [ evaluatedValues ]);
-  const evaluatedValuesChanged = !compare(previousEvaluatedValues, evaluatedValues);
+  const expressionEvaluation = useExpressionEvaluation(valuesExpression);
```
Comment (Skaiir) [inline packages/form-js-viewer/src/render/hooks/useValuesAsync.js:36]: I fixed our hook usage here, getting rid of the condition (hooks should not be conditionally evaluated, this confuses react/preact).

File: packages/form-js-viewer/src/render/hooks/useValuesAsync.js [hunk 2]
```diff
@@ -35,11 +33,8 @@ export default function(field) {
...
+  const evaluatedValues = useDeepCompareState(expressionEvaluation || [], []);
```
Comment (Skaiir) [inline packages/form-js-viewer/src/render/hooks/useValuesAsync.js:37]: I abstracted away all that funky compare logic into its own hook, makes everything cleaner. Plus we may want to re-use this deep comparison stuff other places, for example here:

grafik

**Review by Skaiir — APPROVED** [review]

**Comment by Skaiir** [comment]
@pinussilvestrus Approved by me, but I put in some additional changes. Feel free to merge if you like em.

File: packages/form-js-viewer/src/render/hooks/useValuesAsync.js (hunk 2 above)
Comment (pinussilvestrus) [reply packages/form-js-viewer/src/render/hooks/useValuesAsync.js:37]: Thanks a lot 🏅

**Pull request merged**